cmd/ef_tests/test_runner.rs

```diff
@@ -4,40 +4,43 @@ use crate::types::{Account, TestUnit};
 use ethereum_rust_core::{
     rlp::decode::RLPDecode,
     rlp::encode::RLPEncode,
-    types::{Account as CoreAccount, Block as CoreBlock},
+    types::{Account as CoreAccount, Block as CoreBlock, Transaction as CoreTransaction},
     Address,
 };
 use ethereum_rust_evm::{evm_state, execute_tx, EvmState, SpecId};
 use ethereum_rust_storage::{EngineType, Store};
 
-pub fn execute_test(test: &TestUnit) {
+pub fn execute_test(test_key: &str, test: &TestUnit) {
     // TODO: Add support for multiple blocks and multiple transactions per block.
-    let transaction = test
-        .blocks
-        .first()
-        .unwrap()
-        .transactions
-        .as_ref()
-        .unwrap()
-        .first()
-        .unwrap();
-
-    assert!(execute_tx(
-        &transaction.clone().into(),
-        &test
-            .blocks
-            .first()
-            .as_ref()
-            .unwrap()
-            .block_header
-            .clone()
-            .unwrap()
-            .into(),
-        &mut build_evm_state_from_prestate(&test.pre),
-        SpecId::CANCUN,
-    )
-    .unwrap()
-    .is_success());
+    let block = test.blocks.first().unwrap();
+    let transactions = block.transactions.as_ref().unwrap();
+    for transaction in transactions.iter() {
+        assert_eq!(
+            transaction.clone().sender,
+            CoreTransaction::from(transaction.clone()).sender(),
+            "Expected sender address differs from derived sender address on test: {}",
+            test_key
+        );
+        assert!(
+            execute_tx(
+                &transaction.clone().into(),
+                &test
+                    .blocks
+                    .first()
+                    .as_ref()
+                    .unwrap()
+                    .block_header
+                    .clone()
+                    .unwrap()
+                    .into(),
+                &mut build_evm_state_from_prestate(&test.pre),
```

Comment:
here you are using the same state for each transaction in a block. The n+1 transaction should use the state generated after executing the n transaction. This https://github.com/lambdaclass/ethereum_rust/pull/206 should fix it

Reply:
By moving the evm_state out of the loop this is fixed, now the eip6780_selfdestruct tests which had the "nonce too high" issue are passing.